src/unitxt/operators.py

```diff
@@ -1194,13 +1194,13 @@ def _is_required(self, instance: dict) -> bool:
                 instance_key = dict_get(instance, key)
             except ValueError as ve:
                 raise ValueError(
-                    f"Required filter field ('{key}') in FilterByCondition is not found in {instance}"
+                    f"Required filter field ('{key}') in FilterByCondition is not found in {instance.keys()} of instance"
```

Comment:
It seems that class `FilterByConditionBasedOnFields`, as its name suggests, refers to instance fields only (and not deeper subfields).
If we want to extend the meaning to subfields as well (which I recommend, actually, why not?) then we need to change the above message to:
 f"Required filter field ('{key}') in FilterByCondition is not found in instance" .
If not, then I think we need to replace both `dict_get`-s (`dict_get(instance, key)` and `dict_get(instance, value)`) by a simple, direct lookup:  `instance[key]` and `instance[value]`, and build the error accordingly.